### ISSUE TITLE
[Rosalina][Feature Request] Launch 3dsx processes via set remote exec-file

### ISSUE THREAD

**Issue by Rajveer86** [LumaTeam/Luma3DS#1440]
Looking at e11cc090b27266e922db76cb7a7850d67bda8f18, it seems as though we can start debugging processes that are installed and have a title ID when connected to gdb via `extended-remote`. I would also like the ability to do the same for 3dsx files that are stored on the device.

This would go very well with the integrated support for `remote put`, as we could build a binary, push it to the 3ds and then start debugging with `set remote exec-file`.

If this is already possible and I'm missing something, please let me know :)

**Comment by TuxSH** [comment]
it's kinda hard because you have to deal with APT for app deinit->home menu->app, that's why I've not implemented it.

You could use hbmenu's netloader (press X) with `3dslink`

**Comment by Rajveer86** [comment]
Yeah that's fair enough. I was trying to simplify the development process and find a flow where I could just build, push to the 3ds and start debugging all in one step to make the process as efficient as possible. I'm almost there but the problem I'm facing is that every time I have to re-enable "Force-debug next application at launch" before re-enabling the netloader, and then I can build, push and attach the debugger. It doesn't sound like much, but doing it every time is a bit of a pain point :)

Maybe another way to approach this is to adjust how the "Force-debug next application at launch" works, so that it's always/automatically enabled under certain conditions (e.g. auto on for a specific process name, 3dsx filename e.t.c), and just living with pressing Y everytime? I'm not sure what the easiest way to achieve this would be.

**Comment by SeleDreams** [comment]
I'm having the same issue, i don't know if there has been any progress since then, but this workflow of restarting the homebrew launcher > go to rosaling and enable debug next app at launch all the time is very painful for a development workflow which is why i end up using citra for development at the moment

it would be nice to have something similar to psplink on the psp
